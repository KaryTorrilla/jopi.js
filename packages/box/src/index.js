import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
