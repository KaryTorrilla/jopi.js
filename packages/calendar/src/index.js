import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
