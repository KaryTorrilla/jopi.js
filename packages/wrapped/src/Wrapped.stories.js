import React from 'react'

import { Wrapped } from '.'

export default {
  component: Wrapped,
  title: 'Wrapped',
}

export const WrappedComponent = () => {
  const misItems = ['Pedro', 'Pepito', 'Jose', 'Marcelo', 'Roberto', 'Marian Font', 'Lauti', 'Agus']

  return (
    <Wrapped items={misItems} />
  )
}
