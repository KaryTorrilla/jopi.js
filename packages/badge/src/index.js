import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
