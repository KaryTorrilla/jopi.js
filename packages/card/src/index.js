import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
