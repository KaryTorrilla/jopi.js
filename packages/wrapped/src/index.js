import React, { useEffect, useReducer, useRef, useState } from 'react'
import { Box } from '@oneloop/box'
import { Tags } from '@oneloop/tags'
import { Text } from '@oneloop/text'

const wrappedReducer = (state, action) => {
  switch (action.type) {
    case 'ADD_REST_ITEM':
      return {
        ...state,
        restItems: [...state.restItems, action.payload],
      }
    case 'DELETE_REST_ITEM':
      return {
        ...state,
        restItems: [],
      }
  }
}

export const Wrapped = ({ items, fnClose, widthRestItemsWindow, printKey, tagVariant, idKey = 'id', ...props }) => {
  const [store, dispatch] = useReducer(wrappedReducer, { restItems: [] })
  const [showRestItem, setShowRestItems] = useState(false)
  const Ref = useRef(null)
  const refElements = useRef([])
  const RefItemsWindow = useRef(null)
  let widthWrapper = 0
  let widthElements = 0
  let elements = 0

  useEffect(() => {
    elements = Ref.current.children
    widthWrapper = Ref.current.offsetWidth

    for (let index = 0; index < elements.length; index++) {
      if (!refElements.current.find(elem => elem.id === elements[index].id) && !elements[index].className.includes('numberTag')) {
        refElements.current.push({ ...elements[index], width: elements[index].offsetWidth + 5, id: elements[index].id })
      }

      if (!elements[index].className.includes('numberTag')) {
        widthElements = widthElements + (refElements.current[index].width)
      }

      if (widthElements > (widthWrapper - 30)) {
        elements[index].style.display = elements[index].className.includes('numberTag') ? 'inline-flex' : 'none'

        if (!elements[index].className.includes('numberTag')) {
          if (!store.restItems.find(elem => elem[idKey] === items[index][idKey])) {
            dispatch({ type: 'ADD_REST_ITEM', payload: items[index] })
          }
        }
      }
    }
  }, [items])

  useEffect(() => {
    const handleClick = (event) => {
      if (!RefItemsWindow.current?.contains(event.target) && !event.target.className?.includes('numberTag')) {
        setShowRestItems(false)
      }
    }

    document.addEventListener('click', handleClick)

    return () => {
      document.removeEventListener('click', handleClick)
    }
  }, [showRestItem])

  const handleShowRestItems = () => { setShowRestItems(!showRestItem) }

  const deleteRestItem = (item) => {
    refElements.current = refElements.current.filter(elem => elem.id !== item[idKey])
    dispatch({ type: 'DELETE_REST_ITEM' })
    fnClose(item)
  }

  return (
    <>
      <Box
        __css={{
          display: 'block',
          alignItems: 'center',
          gap: '5px',
          width: '300px',
          overflow: widthWrapper,
          whiteSpace: 'nowrap',
        }}
        {...props}
        ref={Ref}
      >
        {items.map((item, i) => (
          <Tags key={item[idKey] + i} id={item[idKey]} variant={tagVariant} closeIcon={fnClose ? () => deleteRestItem(item) : false} style={{ marginRight: '5px' }}>
            <Text>{item[printKey]}</Text>
          </Tags>
        ))}

        {store.restItems.length > 0
          ? <Tags
            variant={tagVariant}
            className='numberTag'
            style={{ cursor: 'pointer' }}
            onClick={handleShowRestItems}
          >
            +{store.restItems.length}
          </Tags>
          : ''}

      </Box>
      {showRestItem && <Box
        id='restItemsModal'
        ref={RefItemsWindow}
        __css={{
          padding: '12px 14px',
          position: 'absolute',
          top: '25px',
          background: '#FFFFFF',
          boxShadow: '4px 4px 12px rgba(87, 95, 99, 0.1)',
          borderRadius: '8px',
          width: widthRestItemsWindow,
        }}
      >
        {store.restItems.map((item, i) => (
          <Tags key={item[idKey] + i} id={item[idKey]} variant={tagVariant} closeIcon={fnClose ? () => deleteRestItem(item) : false} style={{ marginRight: '5px', marginBottom: '5px' }}>
            <Text>{item[printKey]}</Text>
          </Tags>
        ))}
      </Box>}
    </>
  )
}
