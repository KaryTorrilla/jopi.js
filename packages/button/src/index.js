import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
