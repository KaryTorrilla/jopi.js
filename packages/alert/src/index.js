import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
