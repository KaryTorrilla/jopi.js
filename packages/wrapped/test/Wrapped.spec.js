import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Wrapped } from '../src'

describe('Wrapped', () => {
  test('default', () => {
    const misItems = [
      { id: '1a', neme: 'Pedro' },
      { id: '2b', neme: 'Jose' },
      { id: '3c', neme: 'Juan Carlos' },
      { id: '4d', neme: 'Roberto' },
      { id: '5e', neme: 'Adriano' },
      { id: '6f', neme: 'Roman' },
    ]

    const tree = renderer.create(
      <Wrapped
        items={misItems}
        printKey={'pepito'}
        tagVariant={'secondary'}
      />

    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
