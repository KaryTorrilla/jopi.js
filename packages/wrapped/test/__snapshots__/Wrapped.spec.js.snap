// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapped default 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5px;
  width: 300px;
  overflow: 0;
  white-space: nowrap;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Nunito Sans;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3px 4px;
  gap: 5px;
  height: 20px;
  border-radius: 4px;
  font-weight: 400;
  font-size: 12px;
  line-height: 12px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: primary;
}

<div
  className="c0"
>
  <span
    className="c1"
    id="1a"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  <span
    className="c1"
    id="2b"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  <span
    className="c1"
    id="3c"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  <span
    className="c1"
    id="4d"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  <span
    className="c1"
    id="5e"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  <span
    className="c1"
    id="6f"
    style={
      Object {
        "marginRight": "5px",
      }
    }
  >
    <p
      className="c2"
    />
  </span>
  
</div>
`;
